docs/structured-streaming-programming-guide.md

```diff
@@ -1647,6 +1647,27 @@ For example, sorting on the input stream is not supported, as it requires keepin
 track of all the data received in the stream. This is therefore fundamentally hard to execute 
 efficiently.
 
+### Limitation of global watermark
+
+In some circumstance, some stateful operations could emit rows older than current watermark (with allowed delay),
+which are "late rows" in downstream stateful operations (as Spark uses global watermark) and these rows can be discarded.
+This could bring correctness issue.
+
+This is a limitation of global watermark and operator-wise watermark is not yet supported. Before Spark will support
+operator-wise watermark, Spark will check the logical plan of query and log warning when Spark detects such pattern.
+
+Any stateful operation(s) after any of below stateful operations are possibly having issue:
```

Comment:
any of below -> any of the following
are possibly having issue -> can have this issue